app/src/main/java/com/google/android/gms/samples/wallet/PaymentsUtil.kt

```diff
@@ -0,0 +1,272 @@
+import java.math.RoundingMode
+import java.util.*
+
+/**
+ * Contains helper static methods for dealing with the Payments API.
+ *
+ *
+ * Many of the parameters used in the code are optional and are set here merely to call out their
+ * existence. Please consult the documentation to learn more and feel free to remove ones not
+ * relevant to your implementation.
+ */
+object PaymentsUtil {
+    val MICROS = BigDecimal(1000000.0)
+
+    /**
+     * Create a Google Pay API base request object with properties used in all requests.
+     *
+     * @return Google Pay API base request object.
+     * @throws JSONException
+     */
+    private val baseRequest = JSONObject().put("apiVersion", 2).put("apiVersionMinor", 0)
+
+    /**
+     * Gateway Integration: Identify your gateway and your app's gateway merchant identifier.
+     *
+     *
+     * The Google Pay API response will return an encrypted payment method capable of being charged
+     * by a supported gateway after payer authorization.
+     *
+     *
+     * TODO: Check with your gateway on the parameters to pass and modify them in Constants.java.
+     *
+     * @return Payment data tokenization for the CARD payment method.
+     * @throws JSONException
+     * @see [PaymentMethodTokenizationSpecification](https://developers.google.com/pay/api/android/reference/object.PaymentMethodTokenizationSpecification)
+     */
+    private fun gatewayTokenizationSpecification(): JSONObject {
+        if (Constants.PAYMENT_GATEWAY_TOKENIZATION_PARAMETERS.isEmpty()) {
+            throw RuntimeException(
+                    "Please edit the Constants.java file to add gateway name and other parameters your " + "processor requires")
+        }
+        val tokenizationSpecification = JSONObject()
+
+        tokenizationSpecification.put("type", "PAYMENT_GATEWAY")
+        val parameters = JSONObject(Constants.PAYMENT_GATEWAY_TOKENIZATION_PARAMETERS)
+        tokenizationSpecification.put("parameters", parameters)
+
+        return tokenizationSpecification
+    }
+
+    /**
+     * `DIRECT` Integration: Decrypt a response directly on your servers. This configuration has
+     * additional data security requirements from Google and additional PCI DSS compliance complexity.
+     *
+     *
+     * Please refer to the documentation for more information about `DIRECT` integration. The
+     * type of integration you use depends on your payment processor.
+     *
+     * @return Payment data tokenization for the CARD payment method.
+     * @throws JSONException
+     * @see [PaymentMethodTokenizationSpecification](https://developers.google.com/pay/api/android/reference/object.PaymentMethodTokenizationSpecification)
+     */
+    private fun directTokenizationSpecification(): JSONObject {
+        if (Constants.DIRECT_TOKENIZATION_PARAMETERS.isEmpty()
+                || Constants.DIRECT_TOKENIZATION_PUBLIC_KEY.isEmpty()
+                || Constants.DIRECT_TOKENIZATION_PUBLIC_KEY == null
```

Comment:
`DIRECT_TOKENIZATION_PUBLIC_KEY` is not marked as nullable so this shouldn't be needed here